src/layouts/InterfaceLayout/containers/InteractWithContractContainer/InteractWithContractContainer.vue

```diff
@@ -1,84 +1,349 @@
+    checkType (type) {
+      if (type === 'address' || type === 'string' || type === 'byte') {
+        return 'string'
+      } else {
+        return 'number'
+      }
+    },
+    selectFunction (method) {
+      const contract = new this.$store.state.web3.eth.Contract(
+        [method],
+        this.address
+      )
+      if (method.constant === true && method.inputs.length === 0) {
+        contract.methods[method.name]()
+          .call({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.result = res
+          })
+          .catch(err => {
+            console.log(err)
+          })
+      } else {
+        this.result = ''
+      }
+      this.selectedMethod = method
+    },
+    copyToClipboard (ref) {
+      this.$refs[ref].select()
+      document.execCommand('copy')
+    },
+    deleteInput (ref) {
+      this.$refs[ref].value = ''
+    },
+    switchView (direction) {
+      switch (direction) {
+        case 'forward':
+          this.methods = JSON.parse(this.abi)
+            .map(func => func)
+            .filter(func => func.type !== 'constructor')
+            .filter(func => func.constant !== undefined)
+          this.interact = true
+          break
+        default:
+          this.interact = false
+      }
+    },
+    async write () {
+      const contract = new this.$store.state.web3.eth.Contract([this.selectedMethod], this.address)
+      const params = Object.keys(this.writeInputs).map(input => this.writeInputs[input])
+      this.loading = true
+      if (this.selectedMethod.constant === true) {
+        contract.methods[this.selectedMethod.name](...params)
+          .call({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.result = res
+            this.loading = false
+          })
+          .catch(err => {
+            console.log(err)
+            this.loading = false
+          })
+      } else {
+        this.nonce = await this.$store.state.web3.eth.getTransactionCount(this.$store.state.wallet.getAddressString())
+        this.gasLimit = await contract.methods[this.selectedMethod.name](
+          ...params
+        )
+          .estimateGas({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.transactionFee = unit.fromWei(unit.toWei(this.$store.state.gasPrice, 'gwei') * res, 'ether')
+            return res
+          })
+          .catch(err => console.log(err))
+        this.data = contract.methods[this.selectedMethod.name](...params).encodeABI()
+        this.raw = {
+          from: this.$store.state.wallet.getAddressString(),
+          gas: this.gasLimit,
+          nonce: this.nonce,
+          gasPrice: Number(unit.toWei(this.$store.state.gasPrice, 'gwei')),
+          value: this.value,
+          to: this.address,
+          data: this.data
+        }
+
+        const tx = new EthTx(this.raw)
+        tx.sign(this.$store.state.wallet.getPrivateKey())
```

Comment:
you have to use `web3.eth.sendTransaction` which will sign and send, is this a temp workaround?

Reply:
I tried using that and the send contract method but I'm getting `Error: Returned error: {eth_sendTransaction} Method not found or unavailable`. I also found this: https://github.com/ethereum/web3.js/issues/1637

Reply:
@SteveMieskoski can you look into this?

Reply:
It is happening because `sendTransaction` is directly hitting the node, Steve gotta fix it :)

Reply:
Thanks! I'll wait on @SteveMieskoski on this one then.